File: app/src/main/java/com/guliash/countryquiz/core/AppComponent.java
package com.guliash.countryquiz.core;

import com.guliash.countryquiz.quiz.game.GameModule;
import com.guliash.countryquiz.quiz.provider.QuizProviderModule;
import com.guliash.countryquiz.utils.image.ImageModule;
import com.guliash.countryquiz.utils.network.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, GameModule.class, ImageModule.class, NetworkModule.class,
        QuizProviderModule.class})
public interface AppComponent {

    ActivityComponent plus(ActivityModule activityModule);

}

File: app/src/main/java/com/guliash/quizzes/core/di/AppComponent.kt
package com.guliash.quizzes.core.di

import com.guliash.quizzes.core.di.rx.SchedulersModule
import com.guliash.quizzes.core.repository.di.RepositoryModule
import com.guliash.quizzes.game.di.GameComponent
import com.guliash.quizzes.map.di.MapComponent
import com.guliash.quizzes.map.di.MapModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, SchedulersModule::class, RepositoryModule::class))
interface AppComponent {
    fun plus(): GameComponent

    fun plus(mapModule: MapModule): MapComponent
}
